common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -121,18 +126,25 @@ public MetricSet getAllMetrics() {
   /**
    * Create a {@link TransportClient} connecting to the given remote host / port.
    *
-   * We maintains an array of clients (size determined by spark.shuffle.io.numConnectionsPerPeer)
+   * We maintain an array of clients (size determined by spark.shuffle.io.numConnectionsPerPeer)
    * and randomly picks one to use. If no client was previously created in the randomly selected
    * spot, this function creates a new client and places it there.
    *
+   * We also maintain a last connection failed time of these clients and a fast fail time window
+   * based on io retry wait time. If this connection request can be retried and the last connection
```

Comment:
we should update this since parameter changed.

If the fastFail parameter is true, fail immediately when the last attempt to the same address failed within the fast fail time window (95 percent of the io wait retry timeout). The assumption is the caller will handle retrying.